pandas/core/arrays/base.py

```diff
@@ -657,6 +690,23 @@ def searchsorted(self, value, side="left", sorter=None):
         arr = self.astype(object)
         return arr.searchsorted(value, side=side, sorter=sorter)
 
+    def equals(self, other: ABCExtensionArray) -> bool:
+        """
+        Return if another array is equivalent to this array.
+
+        Parameters
+        ----------
+        other: ExtensionArray
+            Array to compare to this Array.
+
+        Returns
+        -------
+        boolean
+            Whether the arrays are equivalent.
+
+        """
+        return ((self == other) | (self.isna() == other.isna())).all()
```

Comment:
I think this also needs to ensure that the type of `other` matches the type of `self`. We don't want `pd.array([1, 2]).equals(np.array([1, 2]))` to be True (I think).

Reply:
Done.